eng/New-TestResources.ps1

```diff
@@ -122,7 +125,14 @@ if ($TestApplicationId -and !$TestApplicationOid) {
 }
 
 # Format the resource group name based on resource group naming recommendations and limitations.
-$resourceGroupName = "rg-{0}-$baseName" -f ($ServiceDirectory -replace '[\\\/]', '-').Substring(0, [Math]::Min($ServiceDirectory.Length, 90 - $BaseName.Length - 4)).Trim('-')
+$resourceGroupName = if ($CI) {
```

Comment:
Are we using all the digits and alphabets to generate basename. This will be important to have larger key space so we have no conflict when we enable all the OS/languages/version  in pipeline.

Reply:
This is a concern I brought up as well, but we are limited by many different resources (Key Vault being one of them) on the length. While reducing uniqueness of hte otherwise 128-bit integer could result in more collisions, in practice this is pretty rare. If an error occured, it would be obvious (something like "resource group already exists") so it'd be easy to diagnose and just restart a run.

Reply:
Chatted about this with @mikeharder yesterday and we are ok today with the amount of randomness using 16 characters from a guid (formatted without the dashes). This gives us 16^16 names to work with. We could increase this considerably by expanding to the whole alphabet.